### ISSUE TITLE
TSC's helpers cause module to be marked as CommonJS

### ISSUE THREAD

**Issue by devongovett** [parcel-bundler/parcel#8467]
TSC builds helpers that look like this:

```js
var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || ...
```

The `this` reference there causes Parcel to mark the module as CommonJS. This in turn causes the packager to add a setter to the `$parcel$export` call, which may include invalid syntax (e.g. `function (v) { return (parcelRequire("3RBLs")) = v; }`).

https://github.com/parcel-bundler/parcel/blob/3678a9f9ee8f0eb430418734afb3c01d65bdc77a/packages/packagers/js/src/ScopeHoistingPackager.js#L1030-L1032

These are potentially two separate issues:

1. We should detect that the `this` reference is only a read, not a write to avoid marking as CJS.
2. We should detect that the resolved export is a `parcelRequire` and not add a setter in that case?

**Comment by devongovett** [comment]
Filed a TSC issue: https://github.com/microsoft/TypeScript/issues/50761

**Comment by devongovett** [comment]
Fixed by #9318 #9330
